### PR TITLE
Complete catalogue jsx

### PR DESCRIPTION
**Pull request by davidpmccormick**
References #2175 

Renders the `/works` page using Next.js

screen shot 2018-02-26 at 17 59 54

### CODE REVIEW

File: catalogue/webapp/pages/works.js [hunk 1]
```diff
@@ -1,33 +1,39 @@
...
+import {Fragment, Component} from 'react';
+import Router from 'next/router';
 
 type Props = {|
-  values: {| query: string, works: Object |},
-  isSubmitting: boolean,
-  handleChange: () => void,
+  query: {| query?: string, page?: string |},
+  works: {| results: [], totalResults: number |},
+  pagination: Object,
```
Comment (jamesgorrie) [inline catalogue/webapp/pages/works.js]: I thought we had a Pagination type?

File: catalogue/webapp/pages/works.js [hunk 2]
```diff
@@ -57,72 +63,149 @@ const SearchWorks = ({
...
+
+  handleSubmit(event: any) {
+    event.preventDefault();
+
+    const queryString = encodeURIComponent(event.target[0].value);
+
+    // Update the URL, which in turn will update props
+    Router.push({
+      pathname: '/works',
+      query: {query: queryString, page: '1'}
```
Comment (jamesgorrie) [inline catalogue/webapp/pages/works.js]: `page` should be dynamic no?
Also, I wonder if we should be doing state management, this seems like we're hacking the router to deal with our state, and I imagine there might be performance benefits we're losing out on.

File: common/data/series.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  },
+  positionInSeries: 1,
+  series: [{
+    color: 'turquoise',
+    commissionedLength: 1,
+    items: {
+      size: 1
+    }
+  }]
+};
```
Comment (jamesgorrie) [inline common/data/series.js]: Why have we moved this over?
If ew want to - let's clean it up as this has loads of stuff that should be deprecated.

How do we feel about forcing flow on these files too?

File: common/views/components/Pagination/Pagination.js [hunk 4]
```diff
@@ -15,23 +16,79 @@ type Props = {|
...
   </div>
 );
 
 export default Pagination;
+export class PaginationFactory {
+  static fromList(
+    l: [],
+    total: number,
+    currentPage: number = 1,
+    pageSize: number = 32,
```
Comment (jamesgorrie) [inline common/views/components/Pagination/Pagination.js]: https://github.com/wellcometrust/wellcomecollection.org/blob/9b5410114cc635ba3a159beeb4d6524399e6d026/server/services/prismic.js#L51?

**Comment by jamesgorrie** [comment]
An example of using HOCs for state:
https://github.com/wellcometrust/workropper/blob/master/pages/works.js

File: catalogue/webapp/pages/works.js [hunk 5]
```diff
@@ -57,72 +63,149 @@ const SearchWorks = ({
...
-    actions.setValues({ works: json, query });
-    actions.setSubmitting(false);
+class Works extends Component<Props> {
+  handleSubmit: Function;
+  static getInitialProps: Function;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.handleSubmit = this.handleSubmit.bind(this);
```
Comment (jamesgorrie) [inline catalogue/webapp/pages/works.js]: You can avoid doing this by making `handleSubmit` an array function.

File: catalogue/webapp/pages/works.js [hunk 6]
```diff
@@ -57,72 +63,149 @@ const SearchWorks = ({
...
+  }
+
+  handleSubmit(event: any) {
```
Comment (jamesgorrie) [inline catalogue/webapp/pages/works.js]: This is of type `Event` no?

File: catalogue/webapp/pages/works.js (hunk 5 above)
Comment (jamesgorrie) [reply catalogue/webapp/pages/works.js]:
```JS
class Thing {
  handleSubmit: () => {}
}
```

File: catalogue/webapp/pages/works.js [hunk 7]
```diff
@@ -57,72 +63,149 @@ const SearchWorks = ({
...
+      <WorksComponent
+        query={this.props.query}
+        works={this.props.works}
+        pagination={this.props.pagination}
+        handleSubmit={this.handleSubmit} />
+    );
+  }
+}
+
+Works.getInitialProps = async ({ req, query }) => {
```
Comment (jamesgorrie) [inline catalogue/webapp/pages/works.js]: You can put this on the Class itself

File: catalogue/webapp/pages/works.js (hunk 6 above)
Comment (davidpmccormick) [reply catalogue/webapp/pages/works.js]: You're right, it obviously is. The trouble is setting the event parameter to type 'Event' leads to
an 'Indexable signature not found in EventTarget' Flow error. I'm going to set the properties we expect instead (`preventDefault` and `target[0].value`) until we can find a better fix.

File: catalogue/webapp/pages/works.js [hunk 8]
```diff
@@ -58,78 +73,136 @@ const SearchWorks = ({
...
-              <button
-                className='search-box__clear absolute line-height-1 plain-button v-center no-padding js-clear'
-                data-track-event={JSON.stringify({category: 'component', action: 'clear-search:click', label: 'input-id:search-works'})}
-                type='button'>
-                <Icon name='clear' title='clear' />
-              </button>
-            </div>
-            {!query
+            <SearchBox
+              action=''
```
Comment (jamesgorrie) [inline catalogue/webapp/pages/works.js]: Hmmm. Why have it at all?

File: catalogue/webapp/pages/works.js [hunk 9]
```diff
@@ -58,78 +73,136 @@ const SearchWorks = ({
...
+              id='search-works'
```
Comment (jamesgorrie) [inline catalogue/webapp/pages/works.js]: And do we have multiple `SearchBoxes` - just wondering what this is for?

File: server/views/pages/search.njk [hunk 10]
```diff
@@ -120,7 +120,7 @@
           <div class="{{ {s: 12, m: 8, l: 5, xl: 5} | gridClasses }}">
 
             {% set imageModel = henryImage | objectAssign({lazyload: true}) %}
-
+{{ imageModel | dump }}
```
Comment (jamesgorrie) [inline server/views/pages/search.njk]: 👁

**Comment by jamesgorrie** [comment]
@davidpmccormick I added the common lib in the flowconfig so we could stop getting those flow errors. Seems to be working.

https://github.com/wellcometrust/wellcomecollection.org/pull/2270/commits/ab197c63440a47b1ecbfc1c73944ea78bfe7a3b2

**Pull request merged**